forte/data/data_store.py

```diff
@@ -1044,20 +1084,155 @@ def _create_new_entry(
+        of entry.
+
+        - if the entry is of type :class:`~forte.data.ontology.top.Annotation`
+            or :class:`~forte.data.ontology.top.AudioAnnotation`, we assume that
+            `attribute_data` is a list of two elements, indicating the begin and
+            end index of the annotation respectively.
+        - if the entry is of type :class:`~forte.data.ontology.top.Group` or
+            :class:`~forte.data.ontology.top.MultiPackGroup`, we assume that
+            `attribute_data` is a list of one element representing the group's
+            member type.
+        - if the entry is of type :class:`~forte.data.ontology.top.Link` or
+            :class:`~forte.data.ontology.top.MultiPackLink`, we assume that
+            `attribute_data` is a list of two elements representing the link's
+            parent and child type respectively.
+
+        Args:
+            entry (list): The initial version of the data store entry
+                whose values need to be validated.
+
+        Returns:
+            The list that represents the entry with all its values validated
+                and modified (if necessary).
+        """
+        type_name = entry[constants.ENTRY_TYPE_INDEX]
+
+        # Handling creation of annotation type entries
+        if self._is_annotation(type_name):
+            begin_idx = self.get_datastore_attr_idx(
+                type_name, constants.BEGIN_ATTR_NAME
+            )
+            end_idx = self.get_datastore_attr_idx(
+                type_name, constants.END_ATTR_NAME
+            )
+
+            # When creating Annotations, attribute_data has three entries,
+            # representing the begin position, end position and payload index
+            # respectively
+            entry[begin_idx] = attribute_data[0]
+            entry[end_idx] = attribute_data[1]
+
+            # If the begin and end index is not set
+            # Data Store will not be able to add this
+            # entry to the Sorted List (the format in which
+            # annotation are stored)
+            if entry[begin_idx] is None or entry[end_idx] is None:
+                raise KeyError(
+                    "In order to create Annotation or "
+                    "AudioAnnotation, begin and end parameter must be "
+                    "passed in attribute_data. "
+                )
+
+            # If payload is not set, set it to its default value,
+            # ie. 0.
+            payload_id_idx = self.get_datastore_attr_idx(
+                type_name, constants.PAYLOAD_ID_ATTR_NAME
+            )
+
+            if entry[payload_id_idx] is None:
+                entry[payload_id_idx] = 0
+
+        elif any(
+            self._is_subclass(type_name, cls) for cls in (Group, MultiPackGroup)
+        ):
+            members_idx = self.get_datastore_attr_idx(
+                type_name, constants.MEMBER_TID_ATTR_NAME
+            )
+
+            if entry[members_idx] is None:
```

Comment:
Is this already handled by `self._default_attributes_for_type`?

Reply:
Yes it is, I'll remove it